packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts

```diff
@@ -115,287 +406,165 @@ async function setup(
 		},
 	};
 
-	const provider: ITestObjectProvider = getTestObjectProvider();
-	const loader = provider.makeTestLoader(testContainerConfig);
-	const container1 = await createAndAttachContainer(
-		provider.defaultCodeDetails,
-		loader,
-		provider.driver.createCreateNewRequest(provider.documentId),
-	);
-	provider.updateDocumentId(container1.resolvedUrl);
+	async function setup() {
+		const provider: ITestObjectProvider = getTestObjectProvider();
+		const loader = provider.makeTestLoader(testContainerConfig);
+		const container1 = await createAndAttachContainer(
+			provider.defaultCodeDetails,
+			loader,
+			provider.driver.createCreateNewRequest(provider.documentId),
+		);
+		provider.updateDocumentId(container1.resolvedUrl);
+
+		return {
+			loader,
+			provider,
+			container1,
+		};
+	}
 
-	return {
-		loader,
-		provider,
-		container1,
-		testContainerConfig,
-	};
-}
+	/**
+	 * Loads a container and waits for its sequence number to catch up to some existing container (at the time the function begins).
+	 */
+	async function loadAndCatchUpContainer(
+		provider: ITestObjectProvider,
+		existingContainer: IContainer,
+	): Promise<IContainer> {
+		const seq = existingContainer.deltaManager.lastSequenceNumber;
+		const newContainer = await provider.loadTestContainer(testContainerConfig);
+		if (newContainer.deltaManager.lastSequenceNumber < seq) {
```

Comment:
can we just reuse waitContainerToCatchUp

Reply:
yeah, seems like that should suffice here.